src/backend/marsha/core/tests/test_api_liveregistration.py

```diff
@@ -1472,2494 +1377,588 @@ def test_api_liveregistration_create_role_none_token_email_with_user_ok_context_
-        jwt_token.payload["user"] = {
-            "id": "56255f3807599c377bf0e5bf072359fd",
-            "username": "Chachou",
-            "email": None,
-        }
-        response = self.client.get(
-            "/api/liveregistrations/",
-            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
-        )
-
-        self.assertEqual(response.status_code, 200)
-        self.assertEqual(response.json()["count"], 0)
-
-    def test_list_liveregistration_role_student_email_none_consumer_with(
-        self,
-    ):
-        """
-        Student can fetch list requests but will fetch only their registrations.
-
-        Will not receive liveregistrations other than the one corresponding to their token.
-        If token has no email, the user can read liveregistrations of this video for the
-        same duo consumer_site/lti_user_id.
-        """
-        video = VideoFactory(
-            live_state=IDLE,
-            live_type=RAW,
-            starting_at=timezone.now() + timedelta(days=100),
-        )
-        other_video = VideoFactory(
-            live_state=IDLE,
-            live_type=RAW,
-            starting_at=timezone.now() + timedelta(days=100),
-        )
-        other_consumer = ConsumerSiteFactory()
-        # liveregistration for the right video, lti_user_id and consumer_site
-        liveregistration = LiveRegistrationFactory(
-            email="chantal@test-fun-mooc.fr",
-            consumer_site=video.playlist.consumer_site,
-            lti_user_id="56255f3807599c377bf0e5bf072359fd",
-            video=video,
-        )
-        # liveregistration for the same video and lti_user_id but different consumer_site
-        LiveRegistrationFactory(
-            email="chantal2@test-fun-mooc.fr",
-            consumer_site=other_consumer,
-            lti_user_id="56255f3807599c377bf0e5bf072359fd",
-            video=video,
-        )
-        # liveregistration for the same video and consumer_site but different lti_user_id
-        LiveRegistrationFactory(
-            email="chantal3@test-fun-mooc.fr",
-            consumer_site=video.playlist.consumer_site,
-            lti_user_id="DIFFFF3807599c377bf0e5bf072359fd",
-            video=video,
-        )
-        # liveregistration for the same lti_user_id and consumer_site but different video
-        LiveRegistrationFactory(
-            email="chantal2@test-fun-mooc.fr",
-            consumer_site=other_consumer,
-            lti_user_id="56255f3807599c377bf0e5bf072359fd",
-            video=other_video,
-        )
-        # liveregistration for the same video and lti_user_id but no consumer_site
-        LiveRegistrationFactory(
-            email="chantal4@test-fun-mooc.fr",
-            lti_user_id="56255f3807599c377bf0e5bf072359fd",
-            video=video,
+            json.loads(response.content), {"detail": 'Method "PATCH" not allowed.'}
         )
-        # token has context_id and no email
+
+    def test_api_liveregistration_wrong_video_404_create(
```

Comment:
why `create` ? I don't understand. The resource should be created if non existing ? Then why a 404 ?

Reply:
Oh ok, it's about creating the live registration. `test_api_liveregistration_create_with_unknown_video` ?